framework/src/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -896,31 +910,63 @@ protected String dotOutputFileName(UnderlyingAST ast) {
         return null;
     }
 
+    /**
+     * Returns the type of the left-hand side of an assignment without
+     * applying local variable defaults to type variables.
+     *
+     * The type variables that are types of local variables are defaulted to
```

Comment:
Thanks for writing this up, this is very helpful! It would definitely be nice to have this somewhere easier to find than this discussion.
Can the solution for type variables be applied to reference types also? Is the problem that for reference types we don't know what to use instead? Couldn't we just use the default that we would use for parameters/returns/fields then?
I think at least opening an issue for this would be good.